### DISCUSSION TITLE
Can I use VLC to play video on raspberry ?

### DISCUSSION THREAD

**Discussion by bbhxwl** [AvaloniaUI/Avalonia#17339] [Q&A]
Can I use VLC to play video on raspberry ?

**Comment by maxkatz6** [comment]
Probably. It's a wrong repo, we are not working on VLC library.
You should ask in one of these repos with VLC wrappers for Avalonia:
https://code.videolan.org/videolan/LibVLCSharp/
https://github.com/donandren/vlcsharpavalonia

**Comment by maxkatz6** [comment]
Avalonia by itself does work on Raspberry. Still can't say anything about VLC wrapper, which should also support it.

**Comment by bbhxwl** [comment]
Using this project, you only need to install net core to run a simple interface, right?

**Comment by bbhxwl** [comment]
> Probably. It's a wrong repo, we are not working on VLC library.
> You should ask in one of these repos with VLC wrappers for Avalonia:
> https://code.videolan.org/videolan/LibVLCSharp/
> https://github.com/donandren/vlcsharpavalonia

I have tested that the raspberry pie 3B + displays a black screen and does not display video. It is normal on win and MAC

**Comment by bbhxwl** [comment]
In raspberry pie, the normal XAML interface is successfully run, but loading an interface seems to take two or three seconds to display normally?

**Comment by maxkatz6** [comment]
> but loading an interface seems to take two or three seconds to display normally?

It should be faster if you build your app with ReadyToRun enabled, it's a .NET SDK build option, you can find more information about it. There is also possibility of using CoreRT/NativeAOT to speedup startup, but I haven't tried it in long time and not sure how stable it is right now. Usually, ReadyToRun is enough for production apps.

> I have tested that the raspberry pie 3B + displays a black screen and does not display video. It is normal on win and MAC

Try to contact maintainers of your VLC wrapper for more information. As I can see, both wrappers from my previous message can work with Linux, but I have no idea about raspberry, and if they require anything additional for that.

**Comment by bbhxwl** [comment]
I just came into contact with this framework to see if it can use VLC. I tested it successfully, but it takes a long time to start the main interface and pop up the next page in two or three seconds. I don't know where you set 【ReadyToRun 】 and 【CoreRT/NativeAOT】. I'll study it later.

**Comment by bbhxwl** [comment]
[Avalonia] is the code for accessing hardware not cross platform yet? Like recording? Speech recognition?

**Comment by bbhxwl** [comment]
> > but loading an interface seems to take two or three seconds to display normally?
> 
> It should be faster if you build your app with ReadyToRun enabled, it's a .NET SDK build option, you can find more information about it. There is also possibility of using CoreRT/NativeAOT to speedup startup, but I haven't tried it in long time and not sure how stable it is right now. Usually, ReadyToRun is enough for production apps.
> 
> > I have tested that the raspberry pie 3B + displays a black screen and does not display video. It is normal on win and MAC
> 
> Try to contact maintainers of your VLC wrapper for more information. As I can see, both wrappers from my previous message can work with Linux, but I have no idea about raspberry, and if they require anything additional for that.

I found a VLC bug and couldn't solve it.

https://github.com/donandren/vlcsharpavalonia/issues/16
